### PR TITLE
Added Zeeves wallet integration to ZilSwap

### PR DESCRIPTION
**Pull request by whenafirestartstoburn**
Added script reference for https://zeeves.io/sdk/sdk.min.js. The source code of this library is here - https://gitlab.com/whenafirestartstoburn/zeeves-js-auth-sdk.

Demo version with all changes temporary hosted here - https://swap-demo.zeeves.io

Video of how it should work - https://drive.google.com/file/d/19ovyoWbm455P0bpUCpxWLQKYIQjPF2Et/view?usp=sharing

### CODE REVIEW

File: src/app/components/Zeeves/Zeeves.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import React from 'react';
+import { Helmet } from 'react-helmet';
+
+const ZEEVES_HOST = process.env.ZEEVES_HOST || 'zeeves.io';
+
+const Zeeves = () => {
+  return (
+    <Helmet>
+        {/* Zeeves SDK */}
+        <script src={`https://${ZEEVES_HOST}/sdk/sdk.min.js?ts=${new Date().getTime().toString()}`}></script>
```
Comment (RavenXce) [inline src/app/components/Zeeves/Zeeves.js:11]: Is it possible to use a static sdk library that can be properly reviewed? Ideally uploaded to npm.

File: src/app/layouts/WalletDialog/components/ConnectWallet/ConnectWallet.tsx [hunk 2]
```diff
@@ -65,9 +67,13 @@ const ConnectWallet: React.FC<ConnectWalletProps & React.HTMLAttributes<HTMLDivE
...
       <DialogContent className={classes.extraSpacious}>
         <Typography color="textPrimary" variant="body2" align="center">
-          No wallet yet? Download ZilPay
-          {" "}
+          No wallet yet?
+          <br />
+          <br />Download ZilPay{" "}
           <Link rel="noopener noreferrer" target="_blank" href="https://chrome.google.com/webstore/detail/zilpay/klnaejjgbibmhlephnhpmaofohgkpgkd">here</Link>.
+          <br />
+          <br />Or try telegram Zilliqa wallet - {" "}
```
Comment (RavenXce) [inline src/app/layouts/WalletDialog/components/ConnectWallet/ConnectWallet.tsx:75]: Let's use proper branding here?

e.g. "Or try Zeeves, a Telegram based wallet"

File: src/core/wallet/ZilPayConnectedWallet.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ export class ZilPayConnectedWallet implements ConnectedWallet {
   addressInfo: WalletAccountInfo;
 
   constructor(props: ZilPayConnectProps) {
+    console.log('ZilPay props - ' + JSON.stringify(props));
```
Comment (RavenXce) [inline src/core/wallet/ZilPayConnectedWallet.ts:23]: Please remove or use `logger()`.

**Review by RavenXce — CHANGES_REQUESTED** [review]

File: src/core/wallet/ZilPayConnectedWallet.ts (hunk 3 above)
Comment (whenafirestartstoburn) [reply src/core/wallet/ZilPayConnectedWallet.ts]: Removed console.log()

File: src/app/layouts/WalletDialog/components/ConnectWallet/ConnectWallet.tsx (hunk 2 above)
Comment (whenafirestartstoburn) [reply src/app/layouts/WalletDialog/components/ConnectWallet/ConnectWallet.tsx]: Done!

image

File: src/app/components/Zeeves/Zeeves.js (hunk 1 above)
Comment (whenafirestartstoburn) [reply src/app/components/Zeeves/Zeeves.js:11]: I think yea, but let's firstly discuss how do you see it. Personally I think current solution is also okay, but we can discuss.

"zilPay" object in window, for example, is also exported from static js script, which is not immutable and not npm library - inpage.js <script> tag added by ZilPay extension dynamically when page is loaded. You don't install it from npm and it's not the part of your bundle.

image

What happens here is actually the same thing - sdk.min.js exports Zeeves object to window and it is being used the same way as zilPay does.

**Comment by whenafirestartstoburn** [comment]
@RavenXce fixed some critical errors on side of our frontend, deployed new version here - https://swap-demo.zeeves.io - from all latest code, should be okay

**Comment by stevenkhong** [comment]
@whenafirestartstoburn the main issue with directly loading a js file is the risk of malicious code injection, it is extremely hard to ensure the integrity of `zeeves.io/sdk/sdk.min.js` for our users. The js can potentially deface anything on ZilSwap, open a new screen, or run some mining script.

The main difference with ZilPay is that it is loaded from the extension, users can disable it whenever they like or by default easily. However, it will not be as straight forward to to disable a hardcoded js loading from a url. 

The recommendations are:
1. Publish to NPM so the file is fixed by release version.
2. Publish to a Chrome extension and load the SDK from there.

**Comment by whenafirestartstoburn** [comment]
> @whenafirestartstoburn the main issue with directly loading a js file is the risk of malicious code injection, it is extremely hard to ensure the integrity of `zeeves.io/sdk/sdk.min.js` for our users. The js can potentially deface anything on ZilSwap, open a new screen, or run some mining script.
> 
> The main difference with ZilPay is that it is loaded from the extension, users can disable it whenever they like or by default easily. However, it will not be as straight forward to to disable a hardcoded js loading from a url.
> 
> The recommendations are:
> 
> 1. Publish to NPM so the file is fixed by release version.
> 2. Publish to a Chrome extension and load the SDK from there.

@stevenkhong published NPM library - https://www.npmjs.com/package/zeeves-auth-sdk-js. Added this library as a dependency for ZilSwap webapp. Removed static script import.

File: src/app/components/Zeeves/Zeeves.js (hunk 1 above)
Comment (whenafirestartstoburn) [reply src/app/components/Zeeves/Zeeves.js]: made Zeeves an npm library

**Pull request merged**